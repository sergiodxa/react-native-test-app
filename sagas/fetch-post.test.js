import fetchPost from './fetch-post';

const action = {
  type: 'FETCH_POST',
  payload: { id: 1 },
};

describe('fetchPost saga - success', () => {
  let saga;

  beforeAll(() => {
    saga = fetchPost(action);
  });

  it('should return an api call effect', () => {
    const call = saga.next().value;
    expect(call).toMatchSnapshot();
  });

  it('should return a put effect', () => {
    const put = saga.next({
      id: 1,
      userId: 1,
      title: 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
      body: 'quia et suscipit suscipit recusandae consequuntur expedita et cum reprehenderit molestiae ut ut quas totam nostrum rerum est autem sunt rem eveniet architecto',
    }).value;
    expect(put).toMatchSnapshot();
  });

  it('should be done', () => {
    expect(saga.next().done).toBeTruthy();
  });
});

describe('fetchPost saga - failure', () => {
  let saga;

  beforeAll(() => {
    saga = fetchPost(action);
  });

  it('should return an api call effect', () => {
    const call = saga.next().value;
    expect(call).toMatchSnapshot();
  });

  it('should return a put effect', () => {
    const put = saga.next(new Error('test error')).value;
    expect(put).toMatchSnapshot();
  });

  it('should return a delay effect', () => {
    const delay = saga.next().value;
    expect(delay).toMatchSnapshot();
  });

  it('should return a put effect', () => {
    const put = saga.next().value;
    expect(put).toMatchSnapshot();
  });
});
