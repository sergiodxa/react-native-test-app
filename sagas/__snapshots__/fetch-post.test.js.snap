// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchPost saga - failure should return a delay effect 1`] = `undefined`;

exports[`fetchPost saga - failure should return a put effect 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": [Error: test error],
      "type": "ADD_POST",
    },
    "channel": null,
  },
}
`;

exports[`fetchPost saga - failure should return a put effect 2`] = `undefined`;

exports[`fetchPost saga - failure should return an api call effect 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      1,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`fetchPost saga - success should return a put effect 1`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": Object {
        "body": "quia et suscipit suscipit recusandae consequuntur expedita et cum reprehenderit molestiae ut ut quas totam nostrum rerum est autem sunt rem eveniet architecto",
        "id": 1,
        "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
        "userId": 1,
      },
      "type": "ADD_POST",
    },
    "channel": null,
  },
}
`;

exports[`fetchPost saga - success should return an api call effect 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      1,
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
